coalib/results/TextRange.py

```diff
@@ -89,7 +89,31 @@ def end(self):
         return self._end
 
     def overlaps(self, other):
-        return self.start <= other.end and self.end >= other.start
```

Comment:
As discussed earlier on gitter, we should go with the long method I suggested because we need to handle None values properly..